src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -199,70 +209,234 @@ private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarm
 
         if (shouldTryToGetSeed)
         {
-            if (walkToBuilding())
+            final int slot = worker.findFirstSlotInInventoryWith(currentField.getSeed());
+            final ItemStack seeds = new ItemStack(currentField.getSeed());
+            if (slot != -1)
             {
-                return false;
+                requestSeeds = false;
+            }
+            if (!walkToBuilding())
+            {
+                if (isInHut(seeds))
+                {
+                    requestSeeds = false;
+                    isInHut(seeds);
+                }
+                shouldTryToGetSeed = requestSeeds;
+                if (requestSeeds)
+                {
+                    chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", currentField.getSeed().getItemStackDisplayName(seeds));
+                }
             }
-            isInHut(new ItemStack(currentField.getSeed()));
-            shouldTryToGetSeed = false;
         }
 
-        return true;
+        return !shouldTryToGetSeed;
     }
 
     /**
-     * Executes the hoeing of the field.
-     *
-     * @return the next state.
+     * The main work cycle of the Famer.
+     * This checks each block, harvests, tills, and plants.
      */
-    @NotNull
-    private AIState hoe()
+    private AIState cycle()
     {
-        @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+        @Nullable final BuildingFarmer buildingFarmer = getOwnBuilding();
 
-        if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+        if (buildingFarmer == null || checkForHoe() || buildingFarmer.getCurrentField() == null)
         {
             return AIState.PREPARING;
         }
-        @Nullable Field field = getOwnBuilding().getCurrentField();
+
+        @Nullable final Field field = buildingFarmer.getCurrentField();
 
         if (workingOffset != null)
         {
-            BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+            final BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+            // Still moving to the block
             if (walkToBlock(position.up()))
             {
-                return AIState.FARMER_HOE;
+                return AIState.FARMER_WORK;
             }
 
-            if (shouldHoe(position, field))
+            // harvest the block if able to.
+            if (harvestIfAble(position))
+            {
+                setDelay(getLevelDelay());
+            }
+        }
+
+        if (!handleOffsetHarvest(field))
+        {
+            resetVariables();
+            shouldDumpInventory = true;
+            field.setNeedsWork(false);
+            return AIState.IDLE;
+        }
+        return AIState.FARMER_WORK;
+    }
+
+    /**
+     * This (re)initializes a field.
+     * Checks the block above to see if it is a plant, if so, breaks it. Then tills.
+     */
+    private AIState initialize()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 7, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)